devnet-sdk/devstack/devtest/package.go

```diff
@@ -0,0 +1,173 @@
+package devtest
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// P is used by the preset package and system backends as testing interface, to host package-wide resources.
+type P interface {
+	CommonT
+
+	// TempDir creates a temporary directory, and returns the file-path.
+	// This directory is cleaned up at the end of the package,
+	// and can be shared safely between tests that run in that package scope.
+	TempDir() string
+
+	// Cleanup runs the given function at the end of the package-scope.
+	// This function will clean-up once the package-level testing is fully complete.
+	// These resources can thus be shared safely between tests.
+	Cleanup(fn func())
```

Comment:
nit: TempDir and Cleanup are both in `CommonT` already.

Reply:
They're common between the two, but the docs differ, so I repeated them so the docs can be more accurate